src/borg/archiver.py

```diff
@@ -2343,6 +2358,27 @@ def run(self, args):
             logger.warning("Using a pure-python msgpack! This will result in lower performance.")
         return args.func(args)
 
+    def _get_filtered_archives(self, args, manifest):
+        if args.location.archive:
+            logger.error('The options --first and --last can only be used on repository targets.')
+            self.exit_code = EXIT_ERROR
+            return []
+
+        archives = manifest.archives.list()
+        if not archives:
+            logger.critical('There are no archives.')
+            self.exit_code = self.exit_code or EXIT_WARNING
+            return []
+
+        for sortkey in reversed(args.sort_by.split(',')):
+            archives.sort(key=attrgetter(sortkey))
+        if args.last:
+            archives.reverse()
+
+        n = args.first or args.last
```

Comment:
it's seven instead of three.

it's a little less magic, your one-liner relies on short-circuit evaluation.

anyway, take what you like, it was just an option.